components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/graphQL/GraphQLAPIHandler.java

```diff
@@ -159,6 +172,34 @@ public boolean handleRequest(MessageContext messageContext) {
         return false;
     }
 
+    private HashMap<String, Object> convertArrayToMap(ArrayList<OMElement> variablesArray) {
```

Comment:
Change the return type to the generic type. And also when initializing things always use the generic type for the left-hand side, ex  Map<String, Object> externalVariables = new HashMap<>();

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/pull/12441/commits/226c49a261240dc5305f96d69aadb6f86edfae84